src/app/public/modules/description-list/description-list.component.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChildren,
+  ElementRef,
+  HostListener,
+  Input,
+  OnDestroy,
+  QueryList,
+  ViewChild
+} from '@angular/core';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  SkyDescriptionListMode
+} from './types/description-list-mode';
+
+import {
+  SkyDescriptionListOrientation
+} from './types/description-list-orientation';
+
+import {
+  SkyDescriptionListAdapterService
+} from './description-list-adapter-service';
+
+import {
+  SkyDescriptionListContentComponent
+} from './description-list-content.component';
+
+import {
+  SkyDescriptionListService
+} from './description-list.service';
+
+/**
+ * Creates a description list to display term-description pairs.
+ */
+@Component({
+  selector: 'sky-description-list',
+  templateUrl: './description-list.component.html',
+  styleUrls: ['./description-list.component.scss'],
+  providers: [SkyDescriptionListService],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyDescriptionListComponent implements AfterContentInit, OnDestroy {
+
+/**
+ * Specifies a default value to display when no value is provided
+ * for a term-description pair.
+ * @default 'None found'
+ */
+  @Input()
+  public set defaultValue(value: string) {
+    this.descriptionListService.defaultValue.next(value);
+  }
+
+  /**
+   * Specifies the width of the list item when `mode` is set to `nameValuePair`. If not provided,
+   * the list items will have responsive widths at different sizes.
+   */
+  @Input()
+  public listItemWidth: boolean;
```

Comment:
I find `listItem` (and `list item` in the description) a little confusing. What do we mean? At first, I was thinking the problem was that we should change this to `term` because in definition list, this property only affected the label portion, so I thought maybe it was the same here. But when items are stacked with the terms on top of the descriptions, it seems like the width of the term should be the same as the width for the entire term-description pair and also the same as the width for the entire list. If so, can we just change this property to `width` to indicate that it applies to the full list? And change `list item` in the description to `term-description pair` in the description?

Reply:
Sort-of... That is true when the list is vertical, but the main usage is when the list is horizontal. In that scenario, its more clear that the width input is altering the width of the term-description pairs. 

Do you think `width` still makes sense in that case?